sdk/src/runtime/clientNavigation.ts

```diff
@@ -0,0 +1,65 @@
+export function initClientNavigation(
+  opts: {
+    onNavigate: () => void;
+  } = {
+    onNavigate: async function onNavigate() {
+      // @ts-expect-error
+      await globalThis.__rsc_callServer();
+    },
+  },
+) {
+  // Intercept all anchor tag clicks
+  document.addEventListener(
+    "click",
+    async function handleClickEvent(event: MouseEvent) {
+      // should this only work for left click?
+      if (event.button !== 0) {
+        return;
+      }
+
+      if (event.ctrlKey || event.metaKey || event.shiftKey || event.altKey) {
+        return;
+      }
+
+      const target = event.target as HTMLElement;
+      const link = target.closest("a");
+
+      if (!link) {
+        return;
+      }
+
+      const href = link.getAttribute("href");
+      if (!href) {
+        return;
+      }
+
+      // Skip if target="_blank" or similar
+      if (link.target && link.target !== "_self") {
+        return;
+      }
+
+      // Skip if download attribute
+      if (link.hasAttribute("download")) {
```

Comment:
Should we document this?

Reply:
My intuition says that it's such a little footnote that it doesn't really matter. It's the expected behavior.